src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -342,36 +364,192 @@ public static Translation2d[] getModuleTranslations() {
+  }
+
+  public Pose2d getAutoPose() {
+    ChoreoTrajectoryState state =
+        Choreo.getTrajectory("amp 4 local").sample(AutoAim.LOOKAHEAD_TIME);
+
+    return new Pose2d(state.x, state.y, Rotation2d.fromRadians(state.heading));
+  }
+
+  public ChassisSpeeds getAutoSpeeds() {
+    ChoreoTrajectoryState state =
+        Choreo.getTrajectory("amp 4 local").sample(AutoAim.LOOKAHEAD_TIME);
+
+    return new ChassisSpeeds(state.velocityX, state.velocityY, state.angularVelocity);
+  }
+
+  public ChassisSpeeds getFutureSpeeds(boolean isAuto) {
+    return isAuto ? getAutoSpeeds() : getVelocity();
+  }
+
+  public Pose2d getFuturePose(boolean isAuto, ChassisSpeeds speedsFieldRelative) {
+    return isAuto
+        ? getAutoPose()
+        : getLinearFuturePose(AutoAim.LOOKAHEAD_TIME, speedsFieldRelative);
+  }
+
+  /**
+   * Gets the pose at some time in the future, assuming constant velocity
+   *
+   * @param robotRelativeSpeeds the robot relative speed to calculate from
+   * @param time time in seconds
+   * @return The future pose
+   */
+  public Pose2d getLinearFuturePose(double time, ChassisSpeeds speedsFieldRelative) {
+
+    ChassisSpeeds speedsRobotRelative =
+        ChassisSpeeds.fromFieldRelativeSpeeds(speedsFieldRelative, getRotation());
+    return getPose()
+        .transformBy(
+            new Transform2d(
+                speedsRobotRelative.vxMetersPerSecond * time,
+                speedsRobotRelative.vyMetersPerSecond * time,
+                Rotation2d.fromRadians(speedsRobotRelative.omegaRadiansPerSecond * time)));
+  }
+
+  /**
+   * Gets the pose at some time in the future, assuming constant velocity and uses robot's current
+   * speeed
+   *
+   * @param time time in seconds
+   * @return The future pose
+   */
+  public Pose2d getLinearFuturePose(double time) {
+
+    return getPose()
+        .transformBy(
+            new Transform2d(
+                getRobotRelativeSpeeds().vxMetersPerSecond * time,
+                getRobotRelativeSpeeds().vyMetersPerSecond * time,
+                Rotation2d.fromRadians(getRobotRelativeSpeeds().omegaRadiansPerSecond * time)));
+  }
+
+  /**
+   * Gets the pose at some time in the future, assuming constant velocity Uses fixed lookahead time
+   * specified in AutoAim.java
+   *
+   * @return The future pose
+   */
+  @AutoLogOutput(key = "AutoAim/FuturePose")
+  public Pose2d getLinearFuturePose() {
+    return getLinearFuturePose(AutoAim.LOOKAHEAD_TIME);
+  }
+
+  /**
+   * Faces the robot towards a translation on the field
+   *
+   * @param xMetersPerSecond Requested X velocity
+   * @param yMetersPerSecond Requested Y velocity
+   * @param time Time in the future to point from
+   * @return A command refrence that rotates the robot to a computed rotation
+   */
+  public Command pointTowardsTranslationCmd(
+      DoubleSupplier xMetersPerSecond,
+      DoubleSupplier yMetersPerSecond,
+      double time,
+      boolean isAuto) {
```

Comment:
for context the reason there are two methods here is because in teleop we would like to follow a fixed velocity while we prepare to shoot, but in auto we can sample the trajectory we are currently following to get where we will be and what speed we will be at when we will be ready to shoot, so we use that instead.